### PR TITLE
refactor: convert boolean modifiers to enum

### PR DESCRIPTION
**Pull request by midgleyc**
Added some tests for potentially dangerous areas, but I think the coverage is pretty good. It caught a couple of issues, at least!

### CODE REVIEW

**Review by Veracity0 — CHANGES_REQUESTED** [review]
Love this work. Can hardly wait until bitmaps are no longer indexes into array.

File: src/net/sourceforge/kolmafia/Modifiers.java [hunk 1]
```diff
@@ -836,13 +745,17 @@ public boolean set(final Modifiers mods) {
     }
 
     int[] copyBitmaps = mods.bitmaps;
-    for (int index = 0; index < this.bitmaps.length; ++index) {
+    for (int index = 1; index < this.bitmaps.length; ++index) {
```
[the remark this answers is not in the document]
Comment (midgleyc) [reply src/net/sourceforge/kolmafia/Modifiers.java:748]: yes, I was planning to do everything at once. For now it just skips writing into the first index.

**Review by Veracity0 — CHANGES_REQUESTED** [review]

File: src/net/sourceforge/kolmafia/Modifiers.java [hunk 2]
```diff
@@ -146,118 +147,20 @@ public static boolean isNumericModifier(final String key) {
...
-        },
-        Pattern.compile("Attacks Can't Miss")),
-    new BooleanModifier("Pirate", Pattern.compile("Look like a Pirate")),
-    new BooleanModifier("Breakable", Pattern.compile("Breakable")),
-    new BooleanModifier("Drops Items", Pattern.compile("Drops Items")),
-    new BooleanModifier("Drops Meat", Pattern.compile("Drops Meat")),
-  };
-
-  public static final int BOOLEAN_MODIFIERS = Modifiers.booleanModifiers.length;
+  public static final Set<BooleanModifier> BOOLEAN_MODIFIERS =
```
[the remark this answers is not in the document]
Comment (midgleyc) [reply src/net/sourceforge/kolmafia/Modifiers.java:157]: I think I had that originally but it gives a compile error as `Modifiers` needs `BooleanModifiers`, but `BooleanModifiers` needs `Modifiers`, so it's a circular dependency.

[the remark this answers is not in the document]
Comment (midgleyc) [reply src/net/sourceforge/kolmafia/Modifiers.java]: Managed to get it to work after all. Must have been doing something wrong.

[the remark this answers is not in the document]
Comment (Veracity0) [reply src/net/sourceforge/kolmafia/Modifiers.java]: What I mean is, move the declaration of BOOLEAN_MODIFIERS from Modifiers to BooleanModifier.
BooleanModifiers need not import Modifiers; it has its own variable.

Like this:

BooleanModifiers.java:
```
  public static final Set<BooleanModifier> BOOLEAN_MODIFIERS =
      Collections.unmodifiableSet(EnumSet.allOf(BooleanModifier.class));
```
Modifiers.java:
```
import net.sourceforge.kolmafia.modifiers.BooleanModifier;
...
public static final Set<BooleanModifier> BOOLEAN_MODIFIERS = BooleanModifier.BOOLEAN_MODIFIERS;
```

or even:
```
import net.sourceforge.kolmafia.modifiers.BooleanModifier;
import net.sourceforge.kolmafia.modifiers.BooleanModifier.BOOLEAN_MODIFIERS;
```

**Review by Veracity0 — APPROVED** [review]

File: src/net/sourceforge/kolmafia/Modifiers.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Veracity0) [reply src/net/sourceforge/kolmafia/Modifiers.java]: Yes. That is even better. You had to change a few additional files, but it is nice and clean.

**Pull request merged**